List view: Trigger rename block on double click

rinkalpagdar opened the pull request:
Fixes #66829

rinkalpagdar committed:
Block rename on double click

Mamaduka commented:
Thanks for contributing, @rinkalpagdar!

I don't think the proposal for this enhancement has been finalized; there's a possible conflict with another feature request.

I also noticed that the code fails to meet the project coding guidelines and has some leftover debugging code. Please refer to the block editor documentation for coding guidelines and required tools to automate the process:

* https://developer.wordpress.org/block-editor/contributors/code/coding-guidelines/
* https://developer.wordpress.org/block-editor/contributors/code/getting-started-with-code-contribution/

rinkalpagdar committed:
Only rename on duble click if block can be renamed

Mamaduka commented:
@rinkalpagdar, feature development continues in #68744. Please see recent comments - https://github.com/WordPress/gutenberg/pull/68744#issuecomment-2598114528.

rinkalpagdar commented:
@Mamaduka As you suggested, I have corrected my code and fixed the rename block only when it can be renamed. I am also okay with another PR, I just added a demo video here. Thanks!

https://github.com/user-attachments/assets/21f27ed4-0ac3-4642-a6c7-a476b0d66f5c

Mamaduka commented:
Two weeks ago, some improvements were merged for the `BlockRenameModal` modal (#68560), which isn't accounted for here.

I think, at this point, it's better to move forward with #68744.

rinkalpagdar commented:
Okay Please let me know if I need to close this PR.

Mamaduka commented:
I'll close the PR. Please comment on #68744 so that you get contribution props when the feature is merged.

Thank you, @rinkalpagdar!
